esquizito/src/pages/RankingPage.jsx

```diff
@@ -1,88 +1,266 @@
-                <th>{translations.ranking.resultsHeader.successRate}</th>
-                <th>{translations.ranking.resultsHeader.time}</th>
-                {examples.screen.ranking.tableData[0].hits.map((entry) => (
-                  <Box
-                    className='answer'
-                    component='th'
-                    key={`question-${entry.i}`}
-                    width='2%'>
-                    {entry.i}
-                  </Box>
-                ))}
-              </tr>
-            </thead>
-            <tbody>
-              {examples.screen.ranking.tableData.map((entry, i) => (
-                <tr key={entry.name}>
-                  <Box component='td'>{i + 1}</Box>
-                  <Box component='td'>{entry.name}</Box>
-                  <Box component='td' textAlign='center'>
-                    {entry.successRate}%
-                  </Box>
-                  <Box component='td' textAlign='center'>
-                    {entry.time}
-                  </Box>
-                  {entry.hits.map((answer) => (
-                    <td key={`${entry.name}-${answer.i}`}>
-                      <img
-                        src={answer.correct ? correct : incorrect}
-                        alt={
-                          answer.correct
-                            ? translations.ranking.imgDescription.correct
-                            : translations.ranking.imgDescription.incorrect
-                        }
-                      />
-                    </td>
+        {!loading && (
+          <Card>
+            <Table
+              hoverRow
+              sx={{
+                '& img': { width: '100%' },
+                '& thead th': { textAlign: 'center' },
+              }}>
+              <thead>
+                <tr>
+                  <th>
+                    <Link
+                      color='neutral'
+                      component='button'
+                      textColor={orderBy === 'userName' && 'primary.plainColor'}
+                      underline='none'
+                      startDecorator={
+                        <ArrowUpwardRounded
+                          sx={{ opacity: orderBy === 'userName' ? 1 : 0 }}
+                        />
+                      }
+                      sx={{
+                        '& svg': {
+                          transform:
+                            orderBy === 'userName' && order === 'asc'
+                              ? 'rotate(0deg)'
+                              : 'rotate(180deg)',
+                          transition: '0.2s',
+                        },
+                        '&:hover': {
+                          '& svg': { opacity: 1 },
+                        },
+                      }}
+                      onClick={() => sortResultsByKey('userName')}>
+                      {translations.ranking.resultsHeader.name}
+                    </Link>
+                  </th>
```

Comment:
Is it easy to create a sub component for this? Because it seems pretty equal to the others below